examples/js/loaders/OBJLoader.js

```diff
@@ -40,315 +65,431 @@ THREE.OBJLoader.prototype = {
-			var ic = parseVertexIndex( c );
-			var id;
+					this.addVertex( ia, ib, ic );
 
-			if ( d === undefined ) {
+				} else {
 
-				addVertex( ia, ib, ic );
+					id = this.parseVertexIndex( d, vLen );
 
-			} else {
+					this.addVertex( ia, ib, id );
+					this.addVertex( ib, ic, id );
 
-				id = parseVertexIndex( d );
+				}
 
-				addVertex( ia, ib, id );
-				addVertex( ib, ic, id );
+				if ( ua !== undefined ) {
 
-			}
+					var uvLen = this.uvs.length;
 
-			if ( ua !== undefined ) {
+					ia = this.parseUVIndex( ua, uvLen );
+					ib = this.parseUVIndex( ub, uvLen );
+					ic = this.parseUVIndex( uc, uvLen );
 
-				ia = parseUVIndex( ua );
-				ib = parseUVIndex( ub );
-				ic = parseUVIndex( uc );
+					if ( d === undefined ) {
 
-				if ( d === undefined ) {
+						this.addUV( ia, ib, ic );
 
-					addUV( ia, ib, ic );
+					} else {
 
-				} else {
+						id = this.parseUVIndex( ud, uvLen );
 
-					id = parseUVIndex( ud );
+						this.addUV( ia, ib, id );
+						this.addUV( ib, ic, id );
 
-					addUV( ia, ib, id );
-					addUV( ib, ic, id );
+					}
 
 				}
 
-			}
+				if ( na !== undefined ) {
 
-			if ( na !== undefined ) {
+					// Normals are many times the same. If so, skip function call and parseInt.
+					var nLen = this.normals.length;
+					ia = this.parseNormalIndex( na, nLen );
 
-				ia = parseNormalIndex( na );
-				ib = parseNormalIndex( nb );
-				ic = parseNormalIndex( nc );
+					if (na === nb)
+						ib = ia;
+					else
+						ib = this.parseNormalIndex( nb, nLen );
 
-				if ( d === undefined ) {
+					if (na === nc)
+						ic = ia;
+					else
+						ic = this.parseNormalIndex( nc, nLen );
 
-					addNormal( ia, ib, ic );
+					if ( d === undefined ) {
 
-				} else {
+						this.addNormal( ia, ib, ic );
+
+					} else {
+
+						id = this.parseNormalIndex( nd, nLen );
 
-					id = parseNormalIndex( nd );
+						this.addNormal( ia, ib, id );
+						this.addNormal( ib, ic, id );
 
-					addNormal( ia, ib, id );
-					addNormal( ib, ic, id );
+					}
 
 				}
 
+			},
+
+			addLineGeometry : function(vertexes, uvs)
+			{
+				this.object.geometry.type = 'Line';
+
+				var vLen = this.vertices.length;
+				var uvLen = this.uvs.length;
+
+				for (var vi = 0, l = vertexes.length; vi < l; vi++) {
+					this.addVertexLine( this.parseVertexIndex( vertexes[vi], vLen ) );
+				}
+				for (var uvi = 0, l = uvs.length; uvi < l; uvi++) {
+					this.addUVLine( this.parseUVIndex( uvs[uvi], uvLen ) );
+				}
+
 			}
+		};
 
-		}
+		state.startObject('', false);
+		return state;
+	},
+
+	parse: function ( text ) {
 
-		addObject( '' );
+		console.time( 'OBJLoader' );
 
-		// v float float float
-		var vertex_pattern = /^v\s+([\d|\.|\+|\-|e|E]+)\s+([\d|\.|\+|\-|e|E]+)\s+([\d|\.|\+|\-|e|E]+)/;
+		var state = this._createParserState();
 
-		// vn float float float
-		var normal_pattern = /^vn\s+([\d|\.|\+|\-|e|E]+)\s+([\d|\.|\+|\-|e|E]+)\s+([\d|\.|\+|\-|e|E]+)/;
+		if ( text.indexOf('\r\n') !== -1 ) {
```

Comment:
I would actually suspect you want to avoid `text = text.replace('\r\n', '\n');` assignment at all cost here, even if the text is not modified. In a bad case that string can be a 10mb obj file, so I think the less shuffling the better. Again don't know about engine internals but just my intuition that this should be avoided.